pkg/analysis/markerscope/initializer_test.go

```diff
@@ -0,0 +1,350 @@
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With invalid policy", testCase{
+				config: markerscope.MarkerScopeConfig{
+					Policy: "invalid-policy",
+				},
+				expectedErr: `markerscope.policy: Invalid value: "invalid-policy": invalid policy, must be one of: "Warn", "SuggestFix"`,
+			}),
+
+			Entry("With valid marker rules", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With marker rule having empty scope", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{},
+						},
+					},
+				},
+				expectedErr: `scope is required`,
+			}),
+
+			Entry("With marker rule having invalid scope value", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{"invalid"},
+						},
+					},
+				},
+				expectedErr: `invalid scope: "invalid" (must be one of: Field, Type, Any)`,
+			}),
+
+			Entry("With marker rule having invalid schema type", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+							TypeConstraint: &markerscope.TypeConstraint{
+								AllowedSchemaTypes: []markerscope.SchemaType{"invalid-type"},
+							},
+						},
+					},
+				},
+				expectedErr: `invalid type constraint: invalid schema type: "invalid-type"`,
+			}),
+
+			Entry("With valid type constraint with string type", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:string-marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+							TypeConstraint: &markerscope.TypeConstraint{
+								AllowedSchemaTypes: []markerscope.SchemaType{markerscope.SchemaTypeString},
+							},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With valid type constraint with integer type", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:integer-marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+							TypeConstraint: &markerscope.TypeConstraint{
+								AllowedSchemaTypes: []markerscope.SchemaType{markerscope.SchemaTypeInteger},
+							},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With valid type constraint with multiple types", testCase{
```

Comment:
Description doesn't match the test case. It's only checking one allowed schema type and is the same as the case above